### PR TITLE
18 pas form server

### PR DESCRIPTION
**Pull request by allthesignals**
Refactoring code (partly on @godfreyyeung advice) to make controller module own its serialization configuration.

I took this and used an [interceptor](https://docs.nestjs.com/interceptors) to deal with the JSON:API conversion step. Resource-specific configuration is handled on their respective controllers, on interceptor instantiation.

image

### CODE REVIEW

File: server/src/contact/contact.controller.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+@UseInterceptors(new JsonApiSerializeInterceptor('contacts', {
+  attributes: ['contactid', 'emailaddress1'],
+  id: 'contactid',
+}))
+@Controller('contacts')
+export class ContactController {
+  constructor(private readonly contactService: ContactService) {}
+
+  @Get('/')
+  async getUser(@Session() session) {
```
Comment (godfreyyeung) [inline server/src/contact/contact.controller.ts:14]: should be named getContact instead?

File: client/app/authenticators/zap-api-authenticator.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ export default class ZAPAuthenticator extends BaseAuthenticator {
   // a ZAP Token. This request will fail if that cookie doesn't exist
   // https://github.com/simplabs/ember-simple-auth/blob/master/guides/managing-current-user.md#using-a-dedicated-endpoint
   async _fetchUserObject() {
```
Comment (godfreyyeung) [inline client/app/authenticators/zap-api-authenticator.js:30]: should this be named _fetchContactObject() instead?

Comment (allthesignals) [reply client/app/authenticators/zap-api-authenticator.js:30]: yessir! good catch

File: server/src/contact/contact.controller.ts (hunk 1 above)
Comment (allthesignals) [reply server/src/contact/contact.controller.ts:14]: yessir! good catch

**Pull request merged**